ui/e2e/src/pages/Panel.ts

```diff
@@ -43,6 +46,27 @@ export class Panel {
     // The classname selector here is not ideal, but it's all that is available
     // because this lives deeper in another library.
     this.resizeHandle = this.container.locator('..').locator('.react-resizable-handle');
+
+    this.figure = this.container.getByRole('figure');
+    this.loader = this.container.getByLabel('Loading');
+  }
+
+  async isLoaded() {
+    // Wait for the figure to have at least one visible child that is not the loader.
```

Comment:
I tried a few different things before landing on this, and this seems to be the most reliable way to agnostically check if a panel is loaded. Most other cases would have to rely on specific content within the panel, which would be less reusable across tests.

Reply:
Yup! It comes from right here. https://github.com/perses/perses/blob/main/ui/dashboards/src/components/Panel/Panel.tsx#L97